### PR TITLE
Add test and refactor the  AboutPage component

### PR DESCRIPTION
**Pull request by EMaksy**
# Description

Refactor AboutPage component and added multiple components.
Add an AboutPageText component which includes text description.
Add an AboutPageLogo component which includes the Logo.

This will make the AboutPage component better accessible, easier to read and test. 

## How was this tested?

Added test for AboutPage component.
Missing tests for AboutPageText and AboutPageLogo components.
This will be added in a follow-up pr.

### CODE REVIEW

File: assets/js/lib/test-utils/factories.js [hunk 1]
```diff
@@ -87,3 +87,9 @@ export const hostnameFactory = Factory.define(() => ({
   id: faker.datatype.uuid(),
   hostname: faker.hacker.noun(),
 }));
+
+export const aboutPageFactory = Factory.define(() => ({
+  flavor: faker.animal.cat(),
+  sles_subscriptions: faker.datatype.number(),
+  version: faker.git.branch(),
```
Comment (arbulu89) [inline assets/js/lib/test-utils/factories.js:94]: Maybe this better: https://fakerjs.dev/api/system.html#semver

File: assets/js/lib/test-utils/factories.js [hunk 2]
```diff
@@ -87,3 +87,9 @@ export const hostnameFactory = Factory.define(() => ({
   id: faker.datatype.uuid(),
   hostname: faker.hacker.noun(),
 }));
+
+export const aboutPageFactory = Factory.define(() => ({
```
Comment (arbulu89) [inline assets/js/lib/test-utils/factories.js:91]: I would name this `aboutPage`, simply `aboutFactory` as it returns the about data, even though it is only used in the about page by now hehe

File: assets/js/lib/api/about.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import { get } from '@lib/network';
+
+export const getAboutPageData = () => get('/api/about');
```
Comment (arbulu89) [inline assets/js/lib/api/about.js:3]: Maybe better `getAboutData` ?

File: assets/js/components/AboutPage/AboutPage.test.jsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { aboutPageFactory } from '@lib/test-utils/factories';
+import AboutPage from './index';
```
Comment (arbulu89) [inline assets/js/components/AboutPage/AboutPage.test.jsx:4]: You could even to do `import AboutPage from './';` or `import AboutPage from `./AboutPage';`

File: assets/js/components/AboutPage/AboutPage.test.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import { act, screen } from '@testing-library/react';
+import { renderWithRouter } from '@lib/test-utils';
+
+describe('AboutPage component', () => {
+  const apiRequestData = aboutPageFactory.build();
+  it('should render the about page with content from the api', async () => {
+    await act(async () => {
+      renderWithRouter(
```
Comment (arbulu89) [inline assets/js/components/AboutPage/AboutPage.test.jsx:12]: Usually is a good practice to do something like:
```
{ getByText } = renderWithRouter(..
expect(getByText(api..
```
This way you are 100% sure that the used `getByText` is applied in the rendered component

File: assets/js/components/AboutPage/AboutPage.test.jsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    });
+
+    expect(screen.getByText(apiRequestData.flavor)).toBeTruthy();
+    expect(screen.getByText(apiRequestData.version)).toBeTruthy();
+    expect(
+      screen.getByText(`${apiRequestData.sles_subscriptions} found`)
+    ).toBeTruthy();
+  });
+
+  it('should render the about page with state values if api get request fails', async () => {
```
Comment (arbulu89) [inline assets/js/components/AboutPage/AboutPage.test.jsx:24]: Maybe with `default values` ?

**Review by arbulu89 — APPROVED** [review]
Really nice @EMaksy ,
I wrote some cosmetic recommendations, but it is really nice as it is as well

File: assets/js/lib/test-utils/factories.js (hunk 2 above)
Comment (EMaksy) [reply assets/js/lib/test-utils/factories.js:91]: Yeah thanks!
Naming is hard haha

File: assets/js/components/AboutPage/AboutPage.test.jsx (hunk 6 above)
Comment (EMaksy) [reply assets/js/components/AboutPage/AboutPage.test.jsx:24]: Yeah good point

**Pull request merged**